lib/Sema/CSSimplify.cpp

```diff
@@ -8838,6 +8830,19 @@ ConstraintSystem::simplifyApplicableFnConstraint(
   ConstraintLocatorBuilder outerLocator =
     getConstraintLocator(anchor, parts, locator.getSummaryFlags());
 
+  // If the types are obviously equivalent, we're done. This optimization
+  // is not valid for operators though, where an inout parameter does not
+  // have an explicit inout argument.
+  if (type1.getPointer() == desugar2) {
+    if (!isOperator || !hasInOut()) {
+      recordTrailingClosureMatch(
+          getConstraintLocator(
+              outerLocator.withPathElement(ConstraintLocator::ApplyArgument)),
+          TrailingClosureMatching::Forward);
```

Comment:
I'm not super familiar with the trailing closure implementation - does `recordTrailingClosureMatch` just record which trailing closure matching strategy was used for the call?

Reply:
Yeah, `forward` is the default one. It has be to present always otherwise `ExprRewriter::coerceCallArguments` is going to fail here - https://github.com/apple/swift/blob/main/lib/Sema/CSApply.cpp#L5569

Reply:
Gotcha, thanks!